### DISCUSSION TITLE
How to enable locale detection from Accept-Language but disable redirects

### DISCUSSION THREAD

**Discussion by Noitidart** [vercel/next.js#34227] [Help]
I don't want subpath or domain redirection based on locale. But I do need the locale detected and given to each page and each route. Is it possible to do this with middleware?

**Comment by nalejandroveron** [comment]
I'm in the same boat, also needing HTML lang to be populated with no luck

**Reply by Noitidart** [reply]
I also wasn't able to figure it out. I just bought into it and used the subpath redirect.

**Comment by samueldusek** [comment]
I would love to have this option as well.

@Noitidart @Negan1911 Did you make any progress regarding this issue? 🙂

**Reply by Noitidart** [reply]
I ended up using redirects as that's just out of the box and works :(

**Reply by g1na1011** [reply]
@Noitidart Hi! I am working with something similar and would like to redirect the locale subpaths to an error page when they are navigated to. I keep getting an infinite redirecting loop and was wondering if you experienced that? Thanks!

**Reply by Noitidart** [reply]
Hi @g1na1011 - I actually haven't used this yet extensively in production. I can't share much here, but I will maybe in a year or so, so will keep you updated.
